WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -484,14 +486,21 @@ private extension CollectOrderPaymentUseCase {
                                                    onCompletion(.failure(error))
                                                })
             )
-        }
+        },
+                                     emailPromptCompletion: { [weak self] in
+            self?.presentRetryByRestartingError(error: error,
```

Comment:
Yes, this is intentional. It is done to update the currently shown alert after the email prompt is closed. 

After we send a new email receipt, an alert is updated and shows "Email has been sent to {email}" message rather than continue showing "Email receipt" button. 

I'll try to think how to make it clearer. It may be possible to move this call one layer up where `presentRetryByRestartingError` is called rather than calling it within itself.